src/error.rs

```diff
@@ -28,6 +28,14 @@ pub struct Error {
     pub caused_by: ErrorRootCause,
 }
 
+impl std::error::Error for Error {}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{} caused by: {:?}", self.message, self.caused_by)
```

Comment:
This is a bit of a dodgy Display impl because it can include the debug representation of `TryFromIntError` etc, which it probably shouldn't, and if you look below, for most of the variants, `self.message` already contains the display representation of the `caused_by` error. So it may be best to just print out `self.message` (although I don't know if that will actually include all the info).

Reply:
> self.message already contains the display representation of the caused_by error. 

You are right.  I removed the `self.caused_by`.